chalicelib/checks/wrangler_checks.py

```diff
@@ -2270,3 +2276,556 @@ def patch_hic_summary_tables(connection, **kwargs):
+                problem['edge cases'].append([new_record['DCIC UUID'], 'Lab missing 4DN Award'])
+                continue
+            new_record['DCIC Active/Inactive'] = '1'
+            actions['add_excel'].append(new_record)
+
+    # some audit check
+    # check for delete users
+    code_delete_users = list(actions['inactivate_excel'].keys()) + actions['delete_user']
+    if len(excel_delete_users) < len(code_delete_users):
+        diff = [i for i in code_delete_users if i not in excel_delete_users]
+        for i in diff:
+            problem['audit checks'].append([i, 'info in data may not be in sync.'])
+            if i in actions['inactivate_excel'].keys():
+                del actions['inactivate_excel'][i]
+            if i in actions['delete_user']:
+                actions['delete_user'].remove(i)
+
+    # do we need action
+    check.summary = ""
+    for a_key in actions:
+        if actions[a_key]:
+            check.status = 'WARN'
+            check.allow_action = True
+        check.summary += '| {} {}'.format(str(len(actions[a_key])), a_key)
+
+    if problem:
+        num_problems = 0
+        for k in problem.keys():
+            if k != 'cannot find the lab':
+                num_problems += len(problem[k])
+            else:
+                for key in problem[k].keys():
+                    num_problems += len(problem[k][key]['users'])
+
+        check.status = 'WARN'
+        check.summary += '| %s problems' % (str(num_problems))
+
+    check.full_output = {'actions': actions, 'problems': problem}
+    return check
+
+
+@action_function()
+def sync_users_oh_start(connection, **kwargs):
+    action = ActionResult(connection, 'sync_users_oh_start')
+    my_auth = connection.ff_keys
+    sync_users_oh_check_result = action.get_associated_check_result(kwargs).get('full_output', {})
+    actions = sync_users_oh_check_result['actions']
+    user_list = get_oh_google_sheet()
+    # add new users to the data portal
+    if actions.get('add_user'):
+        for a_user in actions['add_user']:
+            del a_user['lab_score']
+            if a_user.get('OH_lab'):
+                del a_user['OH_lab']
+            ff_utils.post_metadata(a_user, 'user', my_auth)
+
+    # Add permissions (lab and awards) to existing users in the data portal
+    if actions.get('add_credentials'):
+        for a_user in actions['add_credentials']:
+            user_uuid = a_user['uuid']
+            del a_user['uuid']
+            del a_user['lab_score']
+            if a_user.get('OH_lab'):
+                del a_user['OH_lab']
+            ff_utils.patch_metadata(a_user, user_uuid, my_auth)
+
+    # remove user's permissions from the data portal
+    if actions.get('delete_user'):
+        for a_user in actions['delete_user']:
+            # Delete the user permissions: submits_for, groups, viewing_groups and lab.
+            ff_utils.patch_metadata({}, a_user, my_auth, add_on='delete_fields=submits_for')
```

Comment:
These 4 patch operations can be merged into a single by using the ff_utils.delete_field function eg.
`ff_utils.delete_field(user_uuid, 'submits_for, lab, viewing_groups, groups', key=my_auth`)